### PR TITLE
TW-2128: Implement QR code to download app

### PR DESCRIPTION
**Pull request by nqhhdev**
## Ticket
- TW-2128

## Resolved

https://github.com/user-attachments/assets/08a9c155-4d2c-4b7f-a210-7e8d9cc647cd

### CODE REVIEW

File: lib/pages/chat_blank/chat_qr_code.dart [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                  child: SvgPicture.asset(
+                    ImagePaths.googlePlay,
+                  ),
+                ),
+                const SizedBox(width: 24),
+                InkWell(
+                  onTap: () {
+                    UrlLauncher(
+                      context,
+                      url: AppConfig.twakeChatGooglePlay,
```
Comment (hoangdat) [inline lib/pages/chat_blank/chat_qr_code.dart:84]: wrong link

File: lib/resource/image_paths.dart [hunk 2]
```diff
@@ -53,6 +53,9 @@ class ImagePaths {
   static String get icPersonCheck => _getImagePath('ic_person_check.svg');
   static String get icTwakeImageLogoBeta =>
       _getImagePath('ic_twake_image_beta.svg');
+  static String get logoPng => _getAssetPath('logo.png');
```
Comment (hoangdat) [inline lib/resource/image_paths.dart:56]: why not use svg?

File: lib/widgets/matrix.dart [hunk 3]
```diff
@@ -91,6 +91,10 @@ class MatrixState extends State<Matrix>
 
   bool waitForFirstSync = false;
 
+  bool firstLogin = false;
+
+  ValueNotifier<bool> showQrCodeDownload = ValueNotifier(false);
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:96]: why need 2 variable?

File: lib/widgets/matrix.dart [hunk 4]
```diff
@@ -830,6 +836,25 @@ class MatrixState extends State<Matrix>
     _contactsManager.reSyncContacts();
   }
 
+  void markFirstLogin() {
+    firstLogin = true;
+  }
+
+  void resetFirstLogin() {
```
[the remark this answers is not in the document]
Comment (nqhhdev) [reply lib/widgets/matrix.dart:843]: Use in here

https://github.com/linagora/twake-on-matrix/blob/7d3c646975292ea8020a094e6f1213e35320061f/lib/pages/chat_blank/chat_qr_code.dart#L51

File: lib/resource/image_paths.dart (hunk 2 above)
Comment (nqhhdev) [reply lib/resource/image_paths.dart:56]: - Reuse old logo already in source code
- Lib not support svg

File: lib/widgets/matrix.dart (hunk 3 above)
Comment (nqhhdev) [reply lib/widgets/matrix.dart:96]: - `firstLogin`: first login recognition
- `showQrCodeDownload`: Show/hide QR code after `BootstrapDialog` closed

**Review by hoangdat — APPROVED** [review]

**Pull request merged**